### PR TITLE
Did resolution returns bbs+ keys

### PR DESCRIPTION
**Pull request by lovesh**

### CODE REVIEW

File: src/modules/did/did.js [hunk 1]
```diff
@@ -795,6 +796,47 @@ class DIDModule {
       }
     });
 
+    // Fetch BBS+ keys if needed
+    if (getBbsPlusSigKeys === true) {
```
Comment (SamHellawell) [inline src/modules/did/did.js:800]: shouldnt need `getBbsPlusSigKeys === true` just `if (getBbsPlusSigKeys)`

File: src/modules/did/did.js [hunk 2]
```diff
@@ -731,12 +732,12 @@ class DIDModule {
...
    */
   // eslint-disable-next-line sonarjs/cognitive-complexity
-  async getDocument(did) {
+  async getDocument(did, { getBbsPlusSigKeys = false } = {}) {
     const hexId = getHexIdentifierFromDID(did);
 
     let didDetails = await this.api.rpc.core_mods.didDetails(hexId, 15);
     if (didDetails.isNone) {
-      throw new NoDIDError(`dock:did:${did}`);
+      throw new NoDIDError(`dock:did:${hexDIDToQualified(hexId)}`);
```
Comment (SamHellawell) [inline src/modules/did/did.js:740]: `dock:did:` should be `did:dock:` ?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@docknetwork/sdk",
-  "version": "2.4.2",
+  "version": "2.5.0",
```
Comment (SamHellawell) [inline package.json:3]: i feel like this change isnt breaking anything so 2.4.3 could work?

File: src/modules/did/did.js [hunk 4]
```diff
@@ -795,6 +796,47 @@ class DIDModule {
       }
     });
 
+    // Fetch BBS+ keys if needed
```
Comment (SamHellawell) [inline src/modules/did/did.js:799]: what would happen if:
- user has three did keys before this change (1, 2 and 3)
- user adds a BBS key

what would the index of the BBS key be? assuming 4, just want to make sure.

if user deletes a key after, would the bbs key still be index 4 or would it go to 3? if it goes to 3, there may be problems

**Review by SamHellawell — CHANGES_REQUESTED** [review]

File: src/modules/did/did.js (hunk 2 above)
Comment (lovesh) [reply src/modules/did/did.js:740]: The argument `did` could be fully-qualified or just hex so can't rely on that.

File: package.json (hunk 3 above)
Comment (lovesh) [reply package.json:3]: As per semver, patch version is for a non-breaking bugfix, non-breaking feature additions need a minor version change.

Comment (SamHellawell) [reply package.json:3]: fair enough, i guess i cheat a little and put some smaller patches into minor bumps. works for me

File: src/modules/did/did.js (hunk 4 above)
Comment (lovesh) [reply src/modules/did/did.js:799]: > assuming 4, just want to make sure.

Yes.

> if user deletes a key after, would the bbs key still be index 4

Remains 4. Key ids don't change and they behaving in the same way for BBS+ as for other types.

File: src/modules/did/did.js (hunk 2 above)
Comment (SamHellawell) [reply src/modules/did/did.js:740]: not sure what you mean, im referring to the error message

File: src/modules/did/did.js (hunk 1 above)
Comment (lovesh) [reply src/modules/did/did.js:800]: Wanted to be explicit

File: src/modules/did/did.js (hunk 4 above)
Comment (SamHellawell) [reply src/modules/did/did.js:799]: sounds good, just wanted to confirm

File: src/modules/did/did.js (hunk 2 above)
Comment (SamHellawell) [reply src/modules/did/did.js:740]: see the edit you made, yeah im aware after a quick search that dock:did is used - could still patch here too

Comment (lovesh) [reply src/modules/did/did.js]: Done.

**Review by SamHellawell — APPROVED** [review]

**Pull request merged**